serverless/internal/storage/fs/fs.go

```diff
@@ -23,29 +23,35 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/golang/glog"
 	"github.com/google/trillian-examples/serverless/api"
 	"github.com/google/trillian-examples/serverless/internal/storage/fs/layout"
 )
 
 const (
 	dirPerm  = 0755
 	filePerm = 0644
+	// TODO(al): consider making immutable files completely readonly
 )
 
 // Storage is a serverless storage implementation which uses files to store tree state.
 // The on-disk structure is:
 //  <rootDir>/leaves/aa/bb/cc/ddeeff...
+//  <rootDir>/leaves/pending/aabbccddeeff...
 //  <rootDir>/seq/aa/bb/cc/ddeeff...
 //  <rootDir>/tile/<level>/aa/bb/ccddee...
 //  <rootDir>/state
 //
 // The functions on this struct are not thread-safe.
 type Storage struct {
 	rootDir string
+	nextSeq uint64
```

Comment:
It would help to comment this with the invariant that it will never point ahead of the next sequence. It may however be pointing at an old entry and this won't cause it to stomp on the already sequenced data. In effect, this is a performance improvement rather than anything functional (if this was always set to 0 the sequencing would work, but it would be slow).

Reply:
done